Update 1.2.1 - 1.2.10 controls (#393)
rancher/rke2: docs/cis_self_assessment.md

**Result:** Pass

**Audit:**
Run the below command on the master node.

```bash
/bin/ps -ef | grep kube-apiserver | grep -v grep
```

> Verify that the result is empty.

Reviewer: ```suggestion
Verify that the `--basic-auth-file` argument does not exist.```

**Result:** Pass

**Remediation:**
By default, RKE2 does not run with basic authentication enabled. No manual remediation is needed.

**Audit:**
Run the below command on the master node.

```bash
/bin/ps -ef | grep kube-apiserver | grep -v grep 
```

> Verify that the result is empty.

Reviewer: ```suggestion
- Verify that the result is empty.
+ Verify that the --basic-auth-file argument does not exist.
```

**Result:** Pass

**Remediation:**
By default, RKE2 kube-apiserver doesn't run with the `--kubelet-https` parameter as it runs with TLS. No manual remediation is needed.

**Audit:**
Run the below command on the master node.

```bash
/bin/ps -ef | grep kube-apiserver | grep -v grep 
```

> Verify that the result is empty.

Reviewer: ```suggestion
Verify that the `--kubelet-https` argument either does not exist or is set to `true`.```

**Result:** Pass

**Audit:**
Run the below command on the master node.

```bash
/bin/ps -ef | grep kube-apiserver | grep -v grep
```

> Verify the output contains these 2 arguments.

Reviewer: What you have is fine, but upsream cis doc has
```
Verify that the --kubelet-client-certificate and --kubelet-client-key arguments exist and they are set as appropriate.
```
I want to stay as aligned to the doc as is reasonable.

**Result:** Pass

**Remediation:**
By default, RKE2 kube-apiserver is ran with this argument for secure communication with kubelet. No manual remediation is needed.

**Audit:**
Run the below command on the master node.

```bash
/bin/ps -ef | grep kube-apiserver | grep -v grep
```

> Verify that the argument is present.

Reviewer: same as above...upstream do is
```
Verify that the `--kubelet-certificate-authority` argument exists and is set as appropriate.
```

Note: This is an Alpha feature in the Kubernetes 1.15 release.
</details>

**Result:** **Not Scored - Operator Dependent**

**Audit:**
Run the below command on the master node.

```bash
/bin/ps -ef | grep kube-apiserver | grep -v grep
```

> Verify that the `--enable-admission-pluginsargument` is set to a value that includes EventRateLimit.

Reviewer: ```suggestion
Verify that the `--enable-admission-plugins` argument is set to a value that includes EventRateLimit.
```

Note: This is an Alpha feature in the Kubernetes 1.15 release.
</details>

**Result:** **Not Scored - Operator Dependent**

**Remediation:**

> By default, RKE2 only sets `NodeRestriction,PodSecurityPolicy` as the parameter to the `--enable-admission-plugins` argument. No manual remediation is needed.

Reviewer: ```suggestion
By default, RKE2 only sets `NodeRestriction,PodSecurityPolicy` as the parameter to the `--enable-admission-plugins` argument.

To configure this, follow the Kubernetes documentation and set the desired limits in a configuration file. Then refer to RKE2's documentation for supply additional api server configuration via the `kube-apiserver-arg` parameter.
```
